Index: pubcookie/src/mod_pubcookie.c
===================================================================
RCS file: /usr/local/cvsroot/webiso/pubcookie/src/mod_pubcookie.c,v
retrieving revision 1.208
diff -u -p -b -w -r1.208 mod_pubcookie.c
--- pubcookie/src/mod_pubcookie.c	15 Aug 2006 05:59:35 -0000	1.208
+++ pubcookie/src/mod_pubcookie.c	16 Aug 2006 10:25:32 -0000
@@ -180,7 +180,8 @@ void dump_server_rec (request_rec * r, p
 		login: %s\n\
 		appsrvid: %s\n\
 		authtype_names: %s\n\
-		use_post: %d\n", scfg->dirdepth, scfg->noblank, (scfg->login == NULL ? "" : scfg->login), (scfg->appsrvid == NULL ? "" : (char *) scfg->appsrvid), (scfg->authtype_names == NULL ? "" : (char *) scfg->authtype_names), scfg->use_post);
+		use_post: %d\n\
+        no_clean_creds: %d\n", scfg->dirdepth, scfg->noblank, (scfg->login == NULL ? "" : scfg->login), (scfg->appsrvid == NULL ? "" : (char *) scfg->appsrvid), (scfg->authtype_names == NULL ? "" : (char *) scfg->authtype_names), scfg->use_post, scfg->no_clean_creds);
 
 }
 
@@ -1432,6 +1433,8 @@ static void *pubcookie_server_merge (poo
     scfg->noblank = nscfg->noblank ? nscfg->noblank : pscfg->noblank;
     /* the following added by ddj@cmu.edu on 2006/05/01 */
     scfg->catenate = nscfg->catenate ? nscfg->catenate : pscfg->catenate;
+    scfg->no_clean_creds =
+        nscfg->no_clean_creds ? nscfg->no_clean_creds : pscfg->no_clean_creds;
     scfg->authtype_names = nscfg->authtype_names ?
         nscfg->authtype_names : pscfg->authtype_names;
     scfg->use_post = nscfg->use_post ? nscfg->use_post : pscfg->use_post;
@@ -2900,6 +2903,20 @@ const char *set_super_debug (cmd_parms *
 
 }
 
+const char *set_no_clean_creds (cmd_parms * cmd, void *mconfig, int flag)
+{
+  server_rec *s = cmd->server;
+  pubcookie_server_rec *scfg;
+  ap_pool *p = cmd->pool;
+
+  scfg = (pubcookie_server_rec *) ap_get_module_config (s->module_config,
+							&pubcookie_module);
+
+  scfg->no_clean_creds = flag;
+
+  return NULL;
+}
+
 /*                                                                            */
 const char *pubcookie_set_no_ssl_ok (cmd_parms * cmd, void *mconfig,
                                      char *v)
@@ -3144,6 +3161,10 @@ static const command_rec pubcookie_comma
                   set_super_debug,
                   NULL, OR_AUTHCFG,
                   "Deprecated, do not use"),
+    AP_INIT_FLAG ("PubCookieNoCleanCreds",
+                  set_no_clean_creds,
+                  NULL, RSRC_CONF,
+                  "Set to leave credentials in place after cleanup"),
 
 /* maybe for future exploration
     AP_INIT_TAKE1("PubCookieNoSSLOK",
@@ -3296,17 +3317,21 @@ static int load_keyed_directives (reques
 static int pubcookie_cleanup (request_rec * r)
 {
     pubcookie_req_rec *rr;
+    pubcookie_server_rec *scfg;
     table *e = r->subprocess_env;
 
     ap_log_rerror (PC_LOG_DEBUG, r, "cleanup");
 
     rr = (pubcookie_req_rec *) ap_get_module_config (r->request_config,
                                                      &pubcookie_module);
+    scfg = (pubcookie_server_rec *) ap_get_module_config (r->server->
+                                                     module_config,
+                                                     &pubcookie_module);
 
     if (!rr)
         return OK;
 
-    if (rr->cred_transfer) {
+    if (rr->cred_transfer && !scfg->no_clean_creds) {
         struct stat sb;
         const char *krb5ccname = ap_table_get (e, "KRB5CCNAME");
 
Index: pubcookie/src/mod_pubcookie.h
===================================================================
RCS file: /usr/local/cvsroot/webiso/pubcookie/src/mod_pubcookie.h,v
retrieving revision 2.21
diff -u -p -b -w -r2.21 mod_pubcookie.h
--- pubcookie/src/mod_pubcookie.h	11 Aug 2006 20:55:57 -0000	2.21
+++ pubcookie/src/mod_pubcookie.h	16 Aug 2006 10:25:32 -0000
@@ -88,6 +88,7 @@ typedef struct
     int dirdepth;
     int noblank;
     int catenate;		/* Added by ddj@cmu.edu on 2006/05/01 */
+    int no_clean_creds;
     char *login;
     unsigned char *appsrvid;
     char *authtype_names;       /* raw arg string from conf */
